clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -27,6 +27,8 @@
  * A key/value pair to be received from Kafka. This also consists of a topic name and 
  * a partition number from which the record is being received, an offset that points 
  * to the record in a Kafka partition, and a timestamp as marked by the corresponding ProducerRecord.
+ *
+ * This class is not thread-safe!
```

Comment:
nit: I think we could highlight it with `<strong>` tag.

Reply:
Agreed. I think it could do with a little bit more explanation, too. The JavaDoc on [`KafkaConsumer`](https://kafka.apache.org/39/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#multithreaded) is pretty extensive, so maybe that could be referenced?

Reply:
There are 3 issues (at least) about the multithreaded issue on ConsumerRecords. Hence, it would be better to document it completely.

Could you please also includes following changes?

1. Highlight the risk of using header in multithread
2. Explain why we don't make this object thread-safe